app/src/main/java/me/rocka/fcitx5test/ui/common/DialogSeekBarPreference.kt

```diff
@@ -0,0 +1,112 @@
+package me.rocka.fcitx5test.ui.common
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.widget.SeekBar
+import androidx.appcompat.app.AlertDialog
+import androidx.preference.Preference
+import androidx.preference.PreferenceManager
+import me.rocka.fcitx5test.R
+import me.rocka.fcitx5test.databinding.DialogSeekBarBinding
+
+/**
+ * Custom preference which represents a seek bar which shows the current value in the summary. The
+ * value can be changed by clicking on the preference, which brings up a dialog which a seek bar.
+ * This implementation also allows for a min / max step value, while being backwards compatible.
+ *
+ * @property defaultValue The default value of this preference.
+ * @property min The minimum value of the seek bar. Must not be greater or equal than [max].
+ * @property max The maximum value of the seek bar. Must not be lesser or equal than [min].
+ * @property step The step in which the seek bar increases per move. If the provided value is less
+ *  than 1, 1 will be used as step.
+ * @property unit The unit to show after the value. Set to an empty string to disable this feature.
+ */
+class DialogSeekBarPreference : Preference {
+    var defaultValue: Int = 0
+    var min: Int = 0
+    var max: Int = 100
+    var step: Int = 1
+    var unit: String = ""
+
+    private val currentValue: Int
+        get() = preferenceDataStore?.getInt(key, defaultValue) ?: defaultValue
+
+    @Suppress("unused")
+    constructor(context: Context) : this(context, null)
+    constructor(context: Context, attrs: AttributeSet?) :
+            this(context, attrs, R.attr.preferenceStyle)
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) :
+            super(context, attrs, defStyleAttr) {
+        onPreferenceClickListener = OnPreferenceClickListener {
+            showSeekBarDialog()
+            true
+        }
+    }
+
+    override fun onAttachedToHierarchy(preferenceManager: PreferenceManager?) {
+        super.onAttachedToHierarchy(preferenceManager)
+        summary = getTextForValue(currentValue)
+    }
+
+    /**
+     * Generates the text for the given [value] and adds the defined [unit] at the end.
+     */
+    private fun getTextForValue(value: Any): String {
+        return if (value !is Int) {
+            "??$unit"
+        } else {
+            value.toString() + unit
+        }
+    }
```

Comment:
Do we have the case that `value` is not an integer?

Reply:
I have no idea, but I think this is a common exception handling since `value` is able to accept `Any` type. Considering `DialogSeekBarPreference` is a common component that will be used anywhere in the future, I do think this is necessary.

Reply:
Here `getTextForValue(value: Int)` is a private method, and we are calling it with `Int` only ... so `Any` seems unnecessary. And I don't think seekbar can handle preference other than `Int`.

Reply:
OK, will fix it soon.